addons/test_mail/tests/test_mail_resend.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import itertools
+
+from odoo.addons.test_mail.tests import common
+from odoo.tests import tagged
+from odoo.tools import mute_logger
+from odoo.tools.safe_eval import safe_eval
+from odoo import api
+
+
+@tagged('resend_test')
+class TestMailResend(common.BaseFunctionalTest, common.MockEmails):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMailResend, cls).setUpClass()
+        userctx = {
+            'no_reset_password': True,
+            'mail_create_nolog': True,
+            'mail_create_nosubscribe': True,
+            'mail_notrack': True,
+        }
+        #Two users
+        cls.user1 = cls.env['res.users'].with_context(userctx).create({
+            'name': 'Employee 1',
+            'login': 'e1',
+            'email': 'e1',  # invalid email
+            'notification_type': 'email',
+            'groups_id': [(6, 0, [cls.env.ref('base.group_user').id])]
+        })
+        cls.user2 = cls.env['res.users'].with_context(userctx).create({
+            'name': 'Employee 2',
+            'login': 'e2',
+            'email': 'e2@example.com',
+            'notification_type': 'email',
+            'groups_id': [(6, 0, [cls.env.ref('base.group_user').id])]
+        })
+        #Two partner
+        cls.partner1 = cls.env['res.partner'].with_context(userctx).create({
+            'name': 'Partner 1',
+            'email': 'p1'  # invalid email
+        })
+        cls.partner2 = cls.env['res.partner'].with_context(userctx).create({
+            'name': 'Partner 2',
+            'email': 'p2@example.com'
+        })
+
+        cls.send_email_counter = int(0)

```

Comment:
Can't you use assertEmails from MockEmails ? Basic use is to check number of sent emails + from / to . Maybe it could help you.